src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -693,22 +689,36 @@ public static void main(String[] args) {
     AbstractAdaptor.main(new DatabaseAdaptor(), args);
   }
   
-  private Acl getAcl(Connection conn, String uniqueId) throws SQLException {
-    try (PreparedStatement stmt = getAclFromDb(conn, uniqueId);
-        ResultSet rs = stmt.executeQuery()) {
-      log.finer("got acl");
-      return buildAcl(rs, aclPrincipalDelimiter, aclNamespace);
+  private Acl getAcl(Connection conn, ResultSet rs, String uniqueId)
+      throws SQLException {
+    if (aclSql != null || rs == null) {
+      try (PreparedStatement stmt = getAclFromDb(conn, uniqueId);
+          ResultSet resSet = stmt.executeQuery()) {
```

Comment:
resSet is a terrible name. It gives no indication of how it's different from rs, is uses a short name for a parameter and a longer name for a local variable, and it uses a non-standard abbreviation. Better would be something like s/rs/contentRs/g and then s/resSet/rs/g.